python/ray/air/_internal/usage.py

```diff
@@ -0,0 +1,18 @@
+from typing import TYPE_CHECKING
+
+from ray._private.usage.usage_lib import TagKey, record_extra_usage_tag
+
+if TYPE_CHECKING:
+    from ray.train.trainer import BaseTrainer
+
+
+def tag_air_trainer(trainer: "BaseTrainer"):
+    from ray.train.trainer import BaseTrainer
+
+    assert isinstance(trainer, BaseTrainer)
+    module_path = trainer.__module__
+    if module_path.startswith("ray.train"):
+        result = f"{trainer.__class__.__name__}"
```

Comment:
This is pretty cool! @jjyao is this safe for whitelisting purposes or do we need to be more explicit?

Reply:
So this makes sure we only record the name of trainers that we create right? I think it's enough. cc @thomasdesr

Let's create a test case to cover that.

Reply:
dude, I actually don't get why we need to do this check.
classes are manually tagged by us. users don't tag custom trainers anyways?

Reply:
nevermind, if we tag BaseTrainer, we need this.
I don't think this check is safe though, like custom trainers can live in ray.train as well.
better to actually create a list of build-in Trainers?